server/src/main/java/io/deephaven/server/plugin/python/CallbackAdapter.java

```diff
@@ -4,18 +4,33 @@
 package io.deephaven.server.plugin.python;
 
 import io.deephaven.plugin.Registration.Callback;
+import io.deephaven.server.plugin.type.JsPluginDistribution;
 import org.jpy.PyObject;
 
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.Objects;
+
 class CallbackAdapter {
 
     private final Callback callback;
 
     public CallbackAdapter(Callback callback) {
-        this.callback = callback;
+        this.callback = Objects.requireNonNull(callback);
     }
 
     @SuppressWarnings("unused")
-    public void registerObjectType(String name, PyObject objectTypeAdapter) {
+    public void registerObjectTypePlugin(String name, PyObject objectTypeAdapter) {
         callback.register(new ObjectTypeAdapter(name, objectTypeAdapter));
     }
+
+    @SuppressWarnings("unused")
+    public void registerJsPlugin(String path, String name, String version, String main) throws IOException {
+        try {
+            callback.register(JsPluginDistribution.of(Path.of(path), name, version, main));
+        } catch (UncheckedIOException e) {
```

Comment:
why force the wrap and unwrap, instead of just declaring the IO ex? Maybe this isnt needed now that register() declares IO ex (not sure, since there are several methods named register() that take a plugin arg...)?

Reply:
So, I think the interfaces presented either need to not have checked exceptions, or to `throw Exception` (as opposed to `IOException` (`io.deephaven.plugin.Registration.Callback#register`) given their generic nature. 

This explicit unwrap is providing slightly better stacktraces for python when something goes wrong, but isn't technically needed.

Reply:
given how irritating py+java stack traces are, it might be better to have an explicit "i should be translated with only a message and a frame or three into python" java exception - the example error for permissions is daunting, esp given that java and py stack traces run in the opposite order.

not to mention if the py code that calls register might have been called by a dh app (java), which could have been started by python (dhaal)

Reply:
I sympathize w/ the struggles of python + java mixed traces. Do we present java exception stack simplification anywhere else?

I worry about the added complexity and fragility of introspecting errors we "expect" to see, at least if we try to solve this on a case-by-case basis.

Reply:
Yeah, this is more of a "the error should be as clean as possible, since our target audience is not python developers who have to deal with java, but js developers who have to deal with python _and_ java, understanding neither".

What i'm asking/hoping for is to deal with expected errors as part of the "state machine" rather than leaving them to "eh, here's an IO error, figure it out"